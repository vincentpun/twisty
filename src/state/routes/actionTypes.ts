export enum RoutesActionsTypes {
  CreateQuery = 'ROUTES_QUERY_CREATE',
  FetchRoutesToken = 'ROUTES_QUERY_FETCH_ROUTES',
  FetchRoutesTokenReceive = 'ROUTES_QUERY_FETCH_ROUTES_TOKEN_RECEIVE',
  PollResult = 'ROUTES_QUERY_POLL_RESULT',
  PollResultReceiveInProgress = 'ROUTES_QUERY_POLL_RESULT_RECEIVE_IN_PROGRESS',
  QuerySuccess = 'ROUTES_QUERY_SUCCESS',
  QueryReceive = 'ROUTES_QUERY_RECEIVE',
  QueryError = 'ROUTES_QUERY_ERROR',
}
