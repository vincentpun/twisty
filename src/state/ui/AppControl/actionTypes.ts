export enum AppControlUIActionTypes {
  Select = 'APP_CONTROL_SELECT',
}
