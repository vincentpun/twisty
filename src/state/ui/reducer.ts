import { combineReducers } from 'redux';

import appControl, { AppControlUIState } from './AppControl/reducer';

export interface UIState {
  appControl: Readonly<AppControlUIState>,
}

export default combineReducers<UIState>({
  appControl,
});
