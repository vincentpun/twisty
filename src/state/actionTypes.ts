import { UIActionTypes } from './ui/actionTypes';
import { RoutesActionsTypes } from './routes/actionTypes';

export type ActionTypes = UIActionTypes | RoutesActionsTypes;
