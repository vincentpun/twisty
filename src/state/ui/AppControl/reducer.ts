import { combineReducers } from 'redux';

import { createAction } from 'src/state/actions';
import { AppControlUIActionTypes } from './actionTypes';
import { AppControlSelection } from './types';
import { Actions } from './actions';

export interface AppControlUIState {
  selection: AppControlSelection;
}

const initialState: AppControlUIState = {
  selection: null,
};

export const selection = (state = initialState.selection, action: Actions) => {
  switch (action.type) {
    case AppControlUIActionTypes.Select:
      return action.payload;
    default:
      return state;
  }
}

export default combineReducers<AppControlUIState>({
  selection,
});
