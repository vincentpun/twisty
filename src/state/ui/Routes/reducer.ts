import { combineReducers } from 'redux';
import { Actions } from 'src/state/routes/actions';
import { RoutesActionsTypes } from '../../routes/actionTypes';

export interface RoutesState {
  query: Readonly<string>;
}

export const query = (state: string = null, action: Actions) => {
  switch (action.type) {
    case RoutesActionsTypes.CreateQuery:
      return action.payload;
    default:
      return state;
  }
};

export default combineReducers({
  query,
});
