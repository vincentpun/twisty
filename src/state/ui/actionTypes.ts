import { AppControlUIActionTypes } from './AppControl/actionTypes';

export enum AppUIActionTypes {
  SwitchMode = 'APP_UI_SWITCH_MODE',
}

export type UIActionTypes = AppUIActionTypes | AppControlUIActionTypes;
