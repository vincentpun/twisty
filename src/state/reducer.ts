import { combineReducers } from 'redux';

import ui, { UIState } from './ui/reducer';
import routes, { RoutesState } from './routes/reducer';

export interface State {
  ui: Readonly<UIState>
  routes: Readonly<RoutesState>,
}

export default combineReducers<State>({
  ui,
  routes,
});
