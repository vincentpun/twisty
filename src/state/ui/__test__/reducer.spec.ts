import * as reducer from '../reducer';
import { AppUIActionTypes } from '../actionTypes';

describe('ui', () => {
  it('matches the snapshot', () => {
    const state = reducer.default(undefined, { type: null });
    expect(state).toMatchSnapshot();
  });

  describe('mode', () => {
    it('matches the snapshot', () => {
      const state = reducer.mode(undefined, { type: null, payload: null });
      expect(state).toMatchSnapshot();
    });

    it('handles AppUIActionTypes.SwitchMode', () => {
      let state = reducer.mode(reducer.AppUIMode.Route, { type: AppUIActionTypes.SwitchMode, payload: reducer.AppUIMode.AppControl });
      expect(state).toEqual(reducer.AppUIMode.AppControl);

      state = reducer.mode(reducer.AppUIMode.AppControl, { type: AppUIActionTypes.SwitchMode, payload: reducer.AppUIMode.Route });
      expect(state).toEqual(reducer.AppUIMode.Route);
    });
  });
});
