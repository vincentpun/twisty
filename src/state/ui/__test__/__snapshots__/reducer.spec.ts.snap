// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui matches the snapshot 1`] = `
Object {
  "appControl": Object {
    "selection": null,
  },
}
`;
